src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx

```diff
@@ -0,0 +1,72 @@
+import { Timeline } from '@konturio/ui-kit';
+import { useAtom } from '@reatom/react';
+import { useCallback, useMemo } from 'react';
+import { eventEpisodesController } from '../../controller';
+import { eventEpisodesModel } from '../../model';
+import type { TimelineProps } from '@konturio/ui-kit';
+import type { Episode } from '~core/types';
+
+type WithRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] };
```

Comment:
`-?` would be awesome to get some explanation or link 😄

Reply:
this is literally what it looks like, it removes the optionality from the parameter (remove "?")
https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers

Reply:
wow, cool! thanks for explanation 🙏